docs/source/user_guide/automl.ipynb

```diff
@@ -142,17 +142,6 @@
     "print(regression_objective.name)"
    ]
   },
-  {
```

Comment:
I still think we should mention data checks in the automl user guide.

If I'm following the intention of the data check actions epic, we're moving from "AutoML will check your data for you" to "Evalml has functionality to check your data and recommend fixes before you start AutoML". 

I think it's important to document the relationship between data checks and automl and that it makes sense to document this relationship here. 

So I think we should:
1. Add back the data checks section (changing it a bit to make it clear that `AutoMLSearch` is not running the checks)
2. Import `DefaultDataChecks` and run on the data. 
3. Say that the results of the data checks don't reveal anything that would cause problems so we can run search. 
4. Link to the data checks user guide page for more information

Let me know what you think!

Reply:
Hmmm, I like how this sounds... my only qualm about this is it would make sense for me that the AutoML user guide currently focuses on the functionality within automl (we currently list out error callbacks, passing in custom pipelines, etc.). This seems more like a flow of our the user should use EvalML.
I guess TLDR is: for what our docs currently look like, I like this change. Maybe this also means we should separate out our docs and not shove everything into the AutoML guide; we could have a intro user guide that steps through the steps a user would take, and then a dedicated AutoML guide that deep-dives into AutoMLSearch specifics.

Will update this, thank you for the great feedback @freddyaboulton!